### ISSUE TITLE
Crashes and doesn't write to error logs when attempting to convert encrypted note from NSX.

### ISSUE THREAD

**Issue by ElmerTucker** [kevindurston21/YANOM-Note-O-Matic#62]
When attempting to convert notebook that contains encrypted note from a .NSX file, the script shuts down and doesn't write to the error log.

**To Reproduce**
1. Export a .NSX file that contains an encrypted note.
2. Place in yanom/data directory
3. Perform the .NSX -> Obsidian quick conversion either using standalone application or CLI.
4. Command Line output shows failed issue, but error log is not written to. (See below for traceback)

**Expected behavior**
Script failure should write to error log before closing down.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Yanom Version 1.3

**Additional context**
Noting that the server based NoteStation export doesn't ask you to decrypt notes for backups, but the desktop version did prompt for the note to be decrypted. When you decrypt the notes, running it through seems to go without errors. 

Processing 'Encrypted Files' Notebook
|⚠                                       | (!) 0/2 [0%] in 0.1s (0.00/s)/s, eta: -)
Time taken: 22.09 seconds
Traceback (most recent call last):
  File "yanom.py", line 121, in <module>
  File "notes_converter.py", line 53, in convert_notes
  File "notes_converter.py", line 114, in convert_nsx
  File "notes_converter.py", line 119, in process_nsx_files
  File "nsx_file_converter.py", line 55, in process_nsx_file
  File "nsx_file_converter.py", line 149, in process_notebooks
  File "sn_notebook.py", line 37, in process_notebook_pages
  File "sn_note_page.py", line 50, in process_note
  File "sn_note_page.py", line 82, in create_attachments
TypeError: 'NoneType' object is not iterable
[39056] Failed to execute script yanom

**Comment by kevindurston21** [comment]
Fixed in commit b4563d7a35adf7257eef38e8250a0813536d40df

Thanks Elmer.

I've had a look into this and the root cause of the error appears to be caused by Note Station.  Testing here it is all OK unless I create a notebook that has an encrypted note in the desktop app and sync that to the web app and then export from the web app.  The attachments are not synced to the diskstation server and this is why we get `TypeError: 'NoneType' object is not iterable` as the raw data in the nsx file is 'null' instead of a list of attachments or an empty list of attachments.

I have added handling for the 'null' data and I was working on a more detailed attachment report for the conversions and so i have added a section listing the note pages where the null attachments are found.
```markdown
# Null file links
The following note pages had their attachment data set as null in the export file as such attachments may be missing
Notebook - "encrypted" - Note page - "a note page - Encrypt
Notebook - "encrypted" - Note page - "encrypted note
```

This change and several others I have been working on are now on the develop branch.  I will add one more change to detect and report an encrypted note exported from the web app and will then build a new release version tomorrow...hopefully.

Kevin


# Long Version
Created a new notebook on desktop app.  Created two note pages, one encrypted, one not encrypted.

When exporting from desktop app get asked to decrypt note. 
If ignore, note is not exported and nsx converts
if provide password 2 notes exported and are converted
If sync this note ook to the web app, viewing the notes neither attachment is actually synced to the server.  Exporting the notebook to nsx from the web app results in the attachment values being set to "null" in the raw note data and no attachment binary files are included in the export.  YANOM crashes when it searches decodes the json and tries to iterate the list of attachments, as the list is not a list it is `None`.
```json
{"attachment":null,"brief":"a note page without encryption","content":"U2FsdGVkX19k4wBCFreLuCJ9v6injbozkAd+Qy7wvG9Urur0m5XtcwZPmSavT9FuhKtuamJTQkKKzM4UZMJfCr3IJoQv/U8FWltIZVCNjEA=","ctime":1626601404,"encrypt":true,"latitude":0,"location":"","longitude":0,"mtime":1626601404,"parent_id":"1026_QP2OC7JO6H4E34D4050NURCASC","source_url":"","tag":["unencrypted"],"title":"a note page - Encrypt"}
{"attachment":null,"brief":"an encrypted note \n\n\n\n","content":"<div>an encrypted note&nbsp;</div><div></div><div></div>","ctime":1626601353,"encrypt":false,"mtime":1626601378,"parent_id":"1026_QP2OC7JO6H4E34D4050NURCASC","tag":["encrypted"],"thumb":null,"title":"encrypted note"}
```
Create new notebook in the web app, create two notes one of which is encrypted.  Export to nsx - not asked for password. 
Raw data file has called names for attachments and an additional flag of `encrypted` is set for the encrypted attachment.

```json
{"attachment":{"_-ExXaV1tOCad2A4jbxw_ZA":{"md5":"27a9aadc878b718331794c8bc50a1b8c","name":"test-page.pdf","size":320357,"type":"binary"}},"brief":"a note without encryption and one attachment\n\n\n\n","content":"<div>a note without encryption and one attachment</div><div></div><div></div>","ctime":1626602209,"encrypt":false,"mtime":1626602309,"parent_id":"1026_NJISPKA74T5H5CHDDTL3VBE4CS","thumb":null,"title":"a clean note"}
{"attachment":{"_ETjlagep_kjXkvB86UX1kA":{"encrypt":true,"md5":"c4ee8b831ad1188509c0f33f0c072af5","name":"example-attachment.pdf","size":14481,"type":"binary"}},"content":"U2FsdGVkX19ovd4UWGeTLiPYyV/Tfr8ZfCgY54IEjfexPGfmnRT6LHFJgheuNTFw","ctime":1626602328,"encrypt":true,"mtime":1626602392,"parent_id":"1026_NJISPKA74T5H5CHDDTL3VBE4CS","title":"an encrypted note"}
```

When converting this web created nsx file YANOM run correctly and gives an error stating a file was not found for the encrypted note (this is the encrypted attachment) This is generated in zip_file_reader.py.
```
on 0: Error - unable to find the file "file_c4ee8b831ad1188509c0f33f0c072af5" in the zip file "/Users/user/PycharmProjects/YANOM-Note-O-Matic/src/data/encrypted-on-web/encrypted-on-web.nsx"

```

The message has  improved by adding an optional source message when calling the zip functions so can pass the note  title.

```
on 0: Warning - For the note "an encrypted note" - unable to find the file "file_c4ee8b831ad1188509c0f33f0c072af5" in the zip file "/Users/user/PycharmProjects/YANOM-Note-O-Matic/src/data/encrypted-on-web/encrypted-on-web.nsx"
```

Also instead of exiting for one missing file YANOM now continues.   The new reporting and improved attachment handling options will also report the file as missing
```markdown
# Missing file links
The following files are linked from note pages but were not located on disk
## For the note: [an-encrypted-note.md](/Users/kevindurston/PycharmProjects/YANOM-Note-O-Matic/src/data/encrypted-on-web-out3/encrypted-on-web/an-encrypted-note.md)
/Users/kevindurston/PycharmProjects/YANOM-Note-O-Matic/src/data/encrypted-on-web-out3/encrypted-on-web/attachments/example-attachment.pdf

```

Added handling for attachment data in nsx set as Null.  No longer causes fatal error.  Summary report now lists any note pages where null was found
```markdown
# Null file links
The following note pages had their attachment data set as null in the export file as such attachments may be missing
Notebook - "encrypted" - Note page - "a note page - Encrypt
Notebook - "encrypted" - Note page - "encrypted note
```

**Comment by kevindurston21** [comment]
HI Elmer,

I'm having problems with Windows running my draft 1.4.  So I have released v1.3.2 put the fix in that.  The log file will log a WARNING for any files where 'null' has been set for attachments in the nsx file.

Cheers,

Kevin

Fixed in v1.3.2
